### PR TITLE
docs: Add general paymaster flow example

### PR DESCRIPTION
**Pull request by Jrigada**
This pull request introduces a new example and documentation for using a general flow paymaster contract in zkSync.

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -39,6 +39,7 @@
 - [Gas Overview](./zksync-specifics/gas.md)
 - [Paymaster Overview](./zksync-specifics/paymaster-overview.md)
 - [Examples](./zksync-specifics/examples/README.md)
+  - [General Flow Paymaster](./zksync-specifics/examples/general-paymaster.md)
```
Comment (elfedy) [inline src/SUMMARY.md:42]:
```suggestion
  - [Paymaster General Flow](./zksync-specifics/examples/general-paymaster.md)
```

`General Flow Paymaster` sounds grammatically weird to me but I may be missing some context on where this comes from.

File: src/zksync-specifics/examples/general-paymaster.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
+## Using the zkUsePaymaster Cheatcode in General Flow Paymaster Contracts
+
+This example covers the use of a general flow paymaster contract.
```
Comment (elfedy) [inline src/zksync-specifics/examples/general-paymaster.md:3]:
```suggestion
This example covers the use of a general flow for a paymaster contract. 
```

File: src/zksync-specifics/examples/general-paymaster.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
+## Using the zkUsePaymaster Cheatcode in General Flow Paymaster Contracts
+
+This example covers the use of a general flow paymaster contract. 
+For this example we will use a contract from the paymaster example repository [here](https://github.com/matter-labs/paymaster-examples).
```
Comment (elfedy) [inline src/zksync-specifics/examples/general-paymaster.md:4]:
```suggestion
For this example we will use the `GaslessPaymaster` contract from the paymaster example repository [here](https://github.com/matter-labs/paymaster-examples).
```

File: src/zksync-specifics/examples/general-paymaster.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+The contract we are going to use is the `GaslessPaymaster` contract.
```
Comment (elfedy) [inline src/zksync-specifics/examples/general-paymaster.md:6]:
```suggestion
```

**Review by elfedy — DISMISSED** [review]

File: src/SUMMARY.md (hunk 1 above)
Comment (Karrq) [reply src/SUMMARY.md:42]: "General Flow" would be the adjective in this case

Comment (Jrigada) [reply src/SUMMARY.md:42]: Yes, in the docs is called that way

File: src/zksync-specifics/examples/general-paymaster.md (hunk 2 above)
Comment (Jrigada) [reply src/zksync-specifics/examples/general-paymaster.md:3]: The "General flow" is how the docs describe it

**Review by elfedy — APPROVED** [review]

**Pull request merged**